File: DSA/DSA06029 SẮP XẾP CHÈN - LIỆT KÊ NGƯỢC.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    vector<int> A(n);
    for (auto &&i : A) cin >> i;
    vector<vector<int>> res;
    vector<int> ans;
    
    for (int i = 0; i < n; i++)
    {
        ans.push_back(A[i]);
        for (int j = 0; j < ans.size(); j++)
        {
            if (ans[j] > A[i])
            {
                swap(ans[j], ans[ans.size()-1]);
            }
        }

        res.push_back(ans);
    }
    
    for (int i = res.size()-1; i >= 0; i--)
    {
        
        cout << "Buoc " << i << ": ";
        for (auto &&k : res[i]) cout << k << " ";
        cout << endl;
    }
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    submit();

    return 0;
}

File: DSA_python/DSA06029 SẮP XẾP CHÈN - LIỆT KÊ NGƯỢC.py
def submit():
    n = int(input())
    A = list(map(int, input().split()))
    res = []
    
    ans = []
    for i in range(n):
        ans.append(A[i])
        for j in range(len(ans)):
            if ans[j] > A[i]:
                ans[j], ans[len(ans)-1] = ans[len(ans)-1], ans[j]

        res.append(ans[:])

    for i in range(len(res)-1, -1, -1):
        print(f"Buoc {i}:", " ".join(map(str, res[i])))


submit()
